### ISSUE TITLE
508-defect-3 [Headings]: Search resources and support articles or all of VA.gov should be a heading 

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#29733]
# [508-defect-3](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-3)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh

## User Story or Problem Statement

As a screen reader user who is used to finding search on the page using headings on Find a Form and Search Results, I expect to be able to do the same on resources and support.

## Details

Since the `h1` for resources and support _does not_ include "Search", folks may not know that a search exists on the page. The search option can be made to be more detectable to screen readers by writing "Search resources and support articles" as a `h2` and then nesting it within a `legend` element.

The search results page is also inconsistent with other search result pages. Search VA.gov's `h1` is "Search VA.gov" and Find a Form's `h1` is "Find a Form"; to maintain consistency and to provide clarity as to _what_ the Veteran is searching for, "Resources and support..." should be included within the `h1` of the results page too.

#### Before
<img width="1373" alt="Screen Shot 2021-09-08 at 11 31 17 AM" src="https://user-images.githubusercontent.com/14154792/132539741-973e8fa5-426b-4759-bfd1-7c2e09207dbb.png">

#### After
<img width="1391" alt="Screen Shot 2021-09-08 at 11 32 27 AM" src="https://user-images.githubusercontent.com/14154792/132539752-dd59920c-348d-4491-b909-669b142b4872.png">

#### Before
![Group 1](https://user-images.githubusercontent.com/14154792/132538956-2ad65027-05af-496d-b300-56540a036404.png)

#### After
![Group 2](https://user-images.githubusercontent.com/14154792/132539257-73cebcac-f7bc-4e06-ac5b-ea1b8fb8743a.png)

## Acceptance Criteria

- [x] On the resources and support page, "Search resources and support articles..." is a `h2` nested within a `legend`
- [x] On the resources and support _search_ page, "Search resources and support articles..." is the `h1`, not "Search results"

## Steps to Recreate

1. Enter `https://www.va.gov/resources/` in browser
2. Open headingsMap
3. Confirm that "Search resources and support..." is not a heading

1. Enter `https://www.va.gov/resources/search/?query=veteran` in browser
2. Confirm `h1` is "Search results"
3. Confirm this is [inconsistent with search VA.gov](https://www.va.gov/search/?query=search+term)

## Proposed Solution

To maintain heading consistency across VA.gov, Find a Form, and Resources and Support, I recommend changing the "Search resources and support..." into a heading nested within a legend. 

## WCAG or Vendor Guidance
[Making labels and legends headings - Gov.uk](https://design-system.service.gov.uk/get-started/labels-legends-headings/)

**Comment by denisecoveyduc** [comment]
@brianalloyd raising for your awareness..

**Comment by brianalloyd** [comment]
@davidconlon @DanielleThierryUSDSVA @bethpottsVADEPO this looks to be more of a product decisions, please review and advise.  Thanks.

**Comment by srsuddath** [comment]
This ticket has been accomplished as part of the following PR
https://github.com/department-of-veterans-affairs/vets-website/pull/19400

**Comment by mgleo** [comment]
Further clarification and ideas from @joshkimux:

> On the sitewide search (results) page, the h1 is simply "Search" without "results" with the first h2 search result heading "Showing x of y"
On resources and support search (results) page, the h1 is "Resources and Support Search Results" with the first h2 being "Search resources and support" and the following h2 being "Showing x of y"
The inconsistency of that search experience is the issue that ticket is addressing. It may be awkward for "search resources and support" to be nested under "resources and support search results."
<img width="758" alt="Screen Shot 2022-03-02 at 7 44 29 PM" src="https://user-images.githubusercontent.com/5553542/157122194-ec471f3b-3f7f-4366-8106-c3af013539b1.png">
<img width="766" alt="Screen Shot 2022-03-02 at 7 44 34 PM" src="https://user-images.githubusercontent.com/5553542/157122205-491f0194-ebbb-44c4-becd-98de0d46c2b7.png">

> I think a possible solution here would be to move "search resources and support or all of [VA.gov](http://va.gov/)" h2 into the h1 to be consistent with [VA.gov](http://va.gov/)'s search, but that may cause IA challenges as it's awkward for the "all of [VA.gov](http://va.gov/)" part to be included in the h1.
I wonder if this might be better handled in the design to provide the option to search all of [VA.gov](http://va.gov/) as a post search option as opposed to a primary one... or if there are other ways of getting around it.
<img width="1118" alt="Screen Shot 2022-03-02 at 7 58 27 PM" src="https://user-images.githubusercontent.com/5553542/157122297-8d802f02-6ce9-4e2a-b32e-1463b4652c32.png">
